File: app/src/main/java/net/radityalabs/contactapp/presentation/util/task/EspressoIdlingResource.java
package net.radityalabs.contactapp.presentation.util.task;

import android.support.test.espresso.IdlingResource;

/**
 * Created by radityagumay on 4/18/17.
 */

public class EspressoIdlingResource {

    private static final String RESOURCE = "GLOBAL";

    private static SimpleCountingIdlingResource mCountingIdlingResource =
            new SimpleCountingIdlingResource(RESOURCE);

    public static void increment() {
        mCountingIdlingResource.increment();
    }

    public static void decrement() {
        mCountingIdlingResource.decrement();
    }

    public static IdlingResource getIdlingResource() {
        return mCountingIdlingResource;
    }
}

File: app/src/main/java/net/radityalabs/contactapp/presentation/util/task/EspressoIdlingResource.kt
package net.radityalabs.contactapp.presentation.util.task

import android.support.test.espresso.IdlingResource

/**
 * Created by radityagumay on 4/18/17.
 */

object EspressoIdlingResource {

    private val RESOURCE = "GLOBAL"

    private val mCountingIdlingResource = SimpleCountingIdlingResource(RESOURCE)

    fun increment() {
        mCountingIdlingResource.increment()
    }

    fun decrement() {
        mCountingIdlingResource.decrement()
    }

    val idlingResource: IdlingResource
        get() = mCountingIdlingResource
}